code/go/0chain.net/chaincore/chain/state/state_context.go

```diff
@@ -51,19 +51,29 @@ func GetMinter(minter ApprovedMinter) (string, error) {
 *    2) The only from clients valid are txn.ClientID and txn.ToClientID (which will be the smart contract's client id)
  */
 
+type CommonStateContextI interface {
```

Comment:
The `common` is a meaningless name, check [package names](https://go.dev/blog/package-names) for "Bad package names" section. It is the same logic as for interface naming. 

Ideally, we should split the `StateContextI` into `state` access and the `chain` access interfaces, they are two different thing. This `CommonStateContextI` is not necessary, we can just use the `QueryStateContextI`, and define it in the `rest` package.